osu.Game/Screens/Play/Player.cs

```diff
@@ -472,6 +472,16 @@ protected virtual ScoreInfo CreateScore()
             else
                 score.User = api.LocalUser.Value;
 
+            try
+            {
+                var performanceCalculator = ruleset?.CreatePerformanceCalculator(Beatmap.Value, score);
+                var totalPp = performanceCalculator?.Calculate(new Dictionary<string, double>());
+                score.PP = totalPp;
+            }
+            catch (Exception)
+            {
+            }
```

Comment:
why is this swallowed? Does it really fail this often to warrant outright silencing any potential exceptions?

Reply:
The try-catch block is only here to wrap calls to Ruleset code (which is user code), so I'm not 100% sure on logging the (very few) potential exceptions here

Reply:
If we're really worried about exposing these exceptions to the end user (which I don't really think we need to be), then this can always be logged at debug level so that the entries don't show up in release builds.

Reply:
IMO it's kind-of weird to actually guard against a calculator..., not to mention that a similar component to this (`DifficultyCalculator`) is used without being wrapped in a try-catch. 

if anything, maybe let the calculator log and early-return if they find something incorrect.

Reply:
the reason for guarding here is that `Ruleset.CreatePerformanceCalculator` may throw `NotImplementedException`; (as it is usercode) we may end up doing this everywhere in the codebase where we call into ruleset code.

Reply:
Guarding against that sounds like caring for a very specific case of failing when it could be easily fixed by the user actually fixing the code by returning null instead. (same can happen with any overridable `CreateXXX` method, you would need to wrap all that in a try-catch as well).

Moreover, afaik throwing a `NotImplementedException` is most likely to remind the developer to implement later on. so it should still crash intentionally. I still can't see a reason this is in a try-catch.

Reply:
> Guarding against that sounds like caring for a very specific case of failing when it could be easily fixed by the user actually fixing the code by returning null instead. (same can happen with any overridable `CreateXXX` method, you would need to wrap all that in a try-catch as well).
>

Wrapping in a try-catch block all interactions with rulesets in the codebase will have to be done at some point in time as we are calling user code and have got little control over what it can do (returning null, throwing exceptions, etc). Let's say I'm doing it there in an anticipated manner.
 
> Moreover, afaik throwing a `NotImplementedException` is most likely to remind the developer to implement later on. so it should still crash intentionally. I still can't see a reason this is in a try-catch.

`NotImplementedException` is part of one of the scenarios of what could go wrong with usercode (throwing exceptions). 


This justifies (at least to me) the need of a try-catch block there